src/targets/mobile/containers/RatingModal.jsx

```diff
@@ -0,0 +1,236 @@
+  sending = false
+
+  async onSubmit (e) {
+    e.preventDefault()
+
+    const { t } = this.context
+    const mailData = {
+      mode: 'from',
+      to: [
+        { name: 'Support', email: FEEDBACK_EMAIL }
+      ],
+      subject: t('mobile.rating.email.subject'),
+      parts: [
+        { type: 'text/plain', body: this.textarea.value.toString() }
+      ]
+    }
+
+    try {
+      this.setState({ sending: true })
+      await cozy.client.jobs.create('sendmail', mailData)
+    } catch (e) {
+      // Sending the email failed; this can happen because of insuficient permissions for example. Not a big deal either in this context.
+    }
+
+    this.setState({ sending: false })
+    this.props.onClose(true)
+  }
+
+  registerElement (element) {
+    this.textarea = element
+  }
+
+  render ({ onClose }, { sending }) {
+    const { t } = this.context
+    return (
+      <Modal
+        title={t('mobile.rating.feedback.title')}
+        withCross={false}
+      >
+        <ModalContent className={styles['feedback-form']}>
+          <form className={'coz-form'} onSubmit={this.onSubmit.bind(this)}>
+            <textarea className={styles['feedback-text']} ref={this.registerElement.bind(this)}>
+              { t('mobile.rating.email.placeholder') }
+            </textarea>
+            <div className={styles['button-block']}>
+              <button className='coz-btn coz-btn--secondary' type='button' onClick={() => onClose(false)}>
+                { t('mobile.rating.feedback.no') }
+              </button>
+              <button className='coz-btn coz-btn--regular' type='submit' aria-busy={sending}>
+                { t('mobile.rating.feedback.yes') }
+              </button>
+            </div>
+          </form>
+        </ModalContent>
+      </Modal>)
+  }
+}
+
+// promptRating is not a component because the native UI is used
+const promptRating = ({ title, message, yes, no, later, softwareName, softwareID }) => {
+  return new Promise((resolve, reject) => {
+    try {
+      AppRate.preferences = {
+        displayAppName: softwareName,
+        inAppReview: true,
+        storeAppURL: {
+          ios: softwareID,
+          android: `market://details?id=${softwareID}`
+        },
+        customLocale: {
+          title,
+          message,
+          cancelButtonLabel: no,
+          laterButtonLabel: later,
+          rateButtonLabel: yes,
+          yesButtonLabel: yes,
+          noButtonLabel: no
+        },
+        callbacks: {
+          onButtonClicked: resolve
+        }
+      }
+
+      AppRate.promptForRating()
+    } catch (e) {
+      reject(e)
+    }
+  })
+}
+
+// Enhance the base component with HOCs
+const withBootDelay = (WrappedComponent, showAfterBoots) => class extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      bootCount: 0,
+      prompted: false
+    }
+  }
+
+  componentStateRestored () {
+    this.setState(prevState => ({
+      bootCount: ++prevState.bootCount
```

Comment:
I cannot understand the name `componentStateRestored` and the implementation that do not take any restored state, but increment the current state.